google/cloud/storage/internal/request_parameters.h

```diff
@@ -21,37 +21,51 @@
 namespace storage {
 inline namespace STORAGE_CLIENT_NS {
 namespace internal {
+// Forward declare the template so we can specialize it first. Defining the
+// specialization first, which is the base class, should be more readable.
+template <typename Parameter, typename... Parameters>
+class RequestParameterList;
+
 /**
  * Refactor common functions that manipulate list of parameters in a request.
  *
- * This class is used in the implementation of `RequestParameters` see below for
- * more details.
+ * This class is used in the implementation of `RequestParameters`, it is the
+ * base class for a hierarchy of `RequestParameterList<Parameters...>` when the
+ * list has a single parameter.
+ *
+ * @tparam Parameter the parameter contained in this object.
  */
-template <typename Parameter, typename... Parameters>
-class RequestParameterList : public RequestParameterList<Parameters...> {
+template <typename Parameter>
+class RequestParameterList<Parameter> {
  public:
-  using RequestParameterList<Parameters...>::set_parameter;
-
   void set_parameter(Parameter&& p) { parameter_ = std::move(p); }
 
   template <typename HttpRequest>
-  void AddToHttpRequest(HttpRequest& request) const {
+  void AddParametersToHttpRequest(HttpRequest &request) const {
     request.AddWellKnownParameter(parameter_);
-    RequestParameterList<Parameters...>::AddToHttpRequest(request);
   }
 
  private:
   Parameter parameter_;
 };
 
-template <typename Parameter>
-class RequestParameterList<Parameter> {
+/**
+ * Refactor common functions that manipulate list of parameters in a request.
+ *
+ * This class is used in the implementation of `RequestParameters` see below for
+ * more details.
```

Comment:
Thanks for the additional comments.  This is way more readable than the first time I looked over it :)